staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -496,6 +505,16 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 		chanSize = 1000
 	}
 
+	// Determine if a consistent stream was requested
+	bookmarkAfterResourceVersion := uint64(0)
+	if watchRV == 0 && opts.ResourceVersionMatch == metav1.ResourceVersionMatchMostRecent && utilfeature.DefaultFeatureGate.Enabled(features.WatchList) {
```

Comment:
It's more a question about the API than the code here - but I would like to allow passing the RV myself too.
In other words, I would like to be able to say that instead of using MostRecent, I would like to use watch-based list but pass that RV myself (e.g. explicitly pass 100).
Will copy the comment to the API one discussion too.

Reply:
pass an RV = 100 and still require `MostRecent` semantics (i.e. consistency) ?

Reply:
No. Pass a "RV=100 and expect a bookmark after I synced to RV=100+ (maybe 100, maybe 101, maybe 110, etc. But I don't care about what the most recent version is).

Basically support "list via watch" for the current "NotOlderThan" semantic.

Reply:
you can do that, in that case, you should not wait for a bookmark event, you will get one after ~60s.

Reply:
But that's a huuuge regression comparing to current "list+watch" pattern. I wouldn't be fine with that as a client.